### PR TITLE
Update to polkadot 0.9.40

### PR DESCRIPTION
**Pull request by krzysztofziobro**
This gives us XCMv3, which might be handy.

Changes based on https://github.com/paritytech/cumulus/tree/polkadot-v0.9.40

### CODE REVIEW

**Review by mike1729 — APPROVED** [review]

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -453,39 +452,33 @@ construct_runtime!(
...
-		System: frame_system::{Pallet, Call, Config, Storage, Event<T>} = 0,
-		ParachainSystem: cumulus_pallet_parachain_system::{
-			Pallet, Call, Config, Storage, Inherent, Event<T>, ValidateUnsigned,
-		} = 1,
-		Timestamp: pallet_timestamp::{Pallet, Call, Storage, Inherent} = 2,
-		ParachainInfo: parachain_info::{Pallet, Storage, Config} = 3,
+		System: frame_system = 0,
+		ParachainSystem: cumulus_pallet_parachain_system = 1,
+		Timestamp: pallet_timestamp = 2,
+		ParachainInfo: parachain_info = 3,
```
Comment (pmikolajczyk41) [inline runtime/src/lib.rs:458]: if im not mistaken, you don't have to specify indices explicitly; or maybe this is requirement for parachains to have strictly matching indices?

File: runtime/src/xcm_config.rs [hunk 2]
```diff
@@ -76,8 +77,9 @@ pub type XcmOriginToTransactDispatchOrigin = (
 
 parameter_types! {
 	// One XCM operation is 1_000_000_000 weight - almost certainly a conservative estimate.
-	pub UnitWeightCost: u64 = 1_000_000_000;
+	pub UnitWeightCost: Weight = Weight::from_parts(1_000_000_000, 64 * 1024);
```
Comment (pmikolajczyk41) [inline runtime/src/xcm_config.rs:80]: what is this 64k?

**Review by pmikolajczyk41 — APPROVED** [review]
it's hard to be review all the changes in detail (with understanding :P), but in general, lgtm

File: runtime/src/xcm_config.rs (hunk 2 above)
Comment (krzysztofziobro) [reply runtime/src/xcm_config.rs:80]: Proof size

File: runtime/src/lib.rs (hunk 1 above)
Comment (krzysztofziobro) [reply runtime/src/lib.rs:458]: Shouldn't matter, but there is a scary comment "Collator support. The order of these 4 are important and shall not change." (which seems strange to me), so I preferred not to touch the indices.

Comment (krzysztofziobro) [reply runtime/src/lib.rs:458]: I you think that it would be cleaner, then I may get rid of them

File: runtime/src/xcm_config.rs (hunk 2 above)
Comment (pmikolajczyk41) [reply runtime/src/xcm_config.rs:80]: is it constant?

Comment (krzysztofziobro) [reply runtime/src/xcm_config.rs:80]: From what I understand it's some kind of a rough upperbound

**Pull request merged**